src/components/schedule-page/table/schedule/schedule-parts/base-cell/base-cell.component.tsx

```diff
@@ -142,12 +154,18 @@ export function BaseCellComponent({
     return "thisMonth";
   }
 
+  function handleClick(): void {
+    if (isComponentVisible) {
+      setIsComponentVisible(false);
+    } else setIsComponentVisible(true);
+  }
   //  #region view
   return (
     <td
-      ref={drop}
+      ref={mergeRefs([ref, drop])}
       className={classNames("mainCell", { selection: isSelected, blocked: isBlocked })}
       id={getId()}
+      onClick={(): void => handleClick()}
```

Comment:
Mamy już w propsach funkcję onClick. Tutaj to handleClick może być mylące moim zdaniem. Można zmienić nazwe na toogleComponentVisibility.
Druga rzecz, że zmieniłbym słowo Component w całości pliku na coś bardziej konkretnego typu: InfoCloud, czy InfoPopper, czy coś takiego.

Reply:
poprawione :)